test/unit/math/prim/arr/err/is_ordered_test.cpp

```diff
@@ -0,0 +1,53 @@
+#include <stan/math/prim/arr.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+#include <string>
+#include <vector>
+
+using stan::math::is_ordered;
+
+TEST(ErrorHandling, isOrdered) {
+  std::vector<double> y_;
+  y_.push_back(0.0);
+  y_.push_back(1.0);
+  y_.push_back(2.0);
+  EXPECT_TRUE(is_ordered(y_));
+
+  y_[2] = std::numeric_limits<double>::infinity();
```

Comment:
I'd much rather see independent tests than ones that modify previous one.  So I'd prefer to see

```
std::vector<double> y = {0, 1, 2};
EXPECT_TRUE(is_ordered(y));

y = { -std::numeric_limits<double>::infinity(), 1, 2};
EXPECT_TRUE(is_ordered(y));
```

and so on.  It makes the tests much more modular and easier to read.